charts/telemetry-collector/templates/configmap.yaml

```
@@ -0,0 +1,78 @@
+{{- include
+      "accelleran.common.configMap"
+      (fromYaml (include "accelleran.telemetry-collector.configMap.args" .))
+-}}
+
+{{- define "accelleran.telemetry-collector.configMap.args" -}}
+{{- $ := . -}}
+
+top:
+  {{ $ | toYaml | nindent 2 }}
+
+name: {{ include "accelleran.common.fullname" (dict "top" $) }}-config
+
+data:
+  config.yaml: |-
+    internal:
+      kafka:
+        url: "{{ include "accelleran.common.bootstrap.kafka.hostname" (dict "top" .) }}:{{ include "accelleran.common.bootstrap.kafka.port" (dict "top" .) }}"
+      {{- if $.Values.config.influxDB }}
+      influxDB:
+        {{- if $.Values.config.influxDB.url }}
+        url: "{{ $.Values.config.influxDB.url }}"
+        {{ else }}
+        url: "http://{{ $.Release.Name }}-influxdb2 }}"
+        {{- end }}
```

I would change it just slightly to not have the condition repeated (and the blocks also both have `url:` repeated).

I would maybe also consider putting this in a template, but probably not necessary at this point.

Is the port also not needed? Or is the default port hidden within the app?

```suggestion
        url: {{ $.Values.config.influxDB.url | default (printf "http://%s-influxdb2 $.Release.Name) | quote }}
```

---

The port is not actually needed - if you don't specify it, it uses the default 8086 which works. But I will add it in case we want to override it.

Actually this whole section is nasty - I found that I also need to included the "token" field, as if it is not specified, the binding code in the app itself does not get the overriden environment variable. As with everything, probably sounds obvious now - but took me a while to figure out why it was not picking up the token.

---

But I have made your suggested change - thanks

---

Actually - coming back to the port - we can just add it to the specified url (eg url: "http://du-influxdb2:8086"), so we do need to do anything there?

---

Yeah, I was just wondering about the port because it was there with kafka but not with influxdb...

What I just realize though is that we shouldn't use that default value. Influxdb isn't deployed as part of this chart, so it should be provided by the user or a chart using it as a dependency (e.g. drax). So we should instead use the `required` method:

```suggestion
        url: {{ $.Values.config.influxDB.url | required "The influxdb URL needs to be provided (config.influxDB.url)" | quote }}
```

---

OK

---

There actually seems to be a problem with this - Test Charts is throwing up an error "helm.go:84: [debug] execution error at (telemetry-collector/templates/configmap.yaml:3:17): The influxdb URL needs to be provided (config.influxDB.url)"

---

This is because in the values file the url is set to "". 
The same issue occurs if we do "helm template --debug charts/telemetry-collector"
We can get around it by setting the url to a random value I think - as it works with helm template debugging when I do that.

---

The error is intentional (and exactly what we want). As we can't provide a proper default for the user at the level of this chart (because we don't include the influxdb2 installation as part of this chart), the user is `required` to do so. We'll eventually do this on the `drax` level as there influxdb2 will be deployed.

---

Because of this we will need to disable the `helm install` tests in the CI for now (until we've found a solution to #162).

```sh
touch charts/telemetry-collector/.ct-install-ignore
```